perfkitbenchmarker/linux_benchmarks/cloudsuite_inmemory_analytics_benchmark.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2015 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs the in-memory analytics benchmark of Cloudsuite.
+
+More info: http://cloudsuite.ch/inmemoryanalytics/
+"""
+
+import re
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+
+flags.DEFINE_string('ima_dataset',
+                    '/data/ml-latest-small',
+                    'Dataset to use for training.')
+flags.DEFINE_string('ima_ratings_file',
+                    '/data/myratings.csv',
+                    'Ratings file to give the recommendation for.')
+FLAGS = flags.FLAGS
+
+BENCHMARK_NAME = 'cloudsuite_inmemory_analytics'
+BENCHMARK_CONFIG = """
+cloudsuite_inmemory_analytics:
+  description: >
+    Run Cloudsuite in-memory analytics benchmark. Specify the number of worker
+    VMs with --num_vms.
+  vm_groups:
+    master:
+      vm_spec: *default_single_core
+    workers:
+      vm_spec: *default_single_core
+      vm_count: 1
+"""
+
+
+def GetConfig(user_config):
+  config = configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+  if FLAGS['num_vms'].present:
+    config['vm_groups']['workers']['vm_count'] = FLAGS.num_vms
+  return config
+
+
+def _HasDocker(vm):
+  resp, _ = vm.RemoteCommand('command -v docker',
+                             ignore_failure=True,
+                             suppress_warning=True)
+  if resp.rstrip() == "":
+    return False
+  else:
+    return True
+
+
+def Prepare(benchmark_spec):
```

Comment:
Yes it can be parallelized per VM. Pulling different docker images within the same VM can also be parallelized, but I'm not sure how to do it in the framework. Maybe just RemoteCommand('docker pull x & docker pull y &')?